phoenix/app/api_products/test/api_products/services/products_service_test.exs

```diff
@@ -0,0 +1,148 @@
+defmodule ApiProducts.ProductsServiceTest do
+  use ApiProducts.DataCase
+
+  import Mock
+
+  alias ApiProducts.ProductsService
+  alias ApiProducts.ElasticService
+  alias ApiProducts.RedisService
+  alias ApiProducts.Management
+
+  setup_all do
+    list_valid = %{"name" => "Nome-teste"}
+
+    create_params = %{
+      "id" => "11aaaa1111111a111111aa11",
+      "sku" => "valid",
+      "name" => "Nome-teste",
+      "description" => "Descrição Teste",
+      "amount" => 100,
+      "price" => 99.5,
+      "barcode" => "030105092"
+    }
+
+    new_valid_params = %{"name" => "produto-atualizado", "amount" => 10, "price" => 19}
+
+    invalid_attrs = %{
+      "sku" => "invalid_*",
+      "name" => "",
+      "description" => "Descrição Teste",
+      "amount" => 10,
+      "price" => 0,
+      "barcode" => "005"
+    }
+
+    [
+      new_valid_params: new_valid_params,
+      list_valid: list_valid,
+      create_params: create_params,
+      invalid_attrs: invalid_attrs
+    ]
+  end
+
+  setup do
+    {:ok, product} =
+      Management.create_product(%{
+        sku: "valid",
+        name: "Nome-teste",
+        description: "Descrição Teste",
+        amount: 100,
+        price: 99.5,
+        barcode: "030105092"
+      })
+
+    [product: product]
+  end
+
+  describe "list/1" do
+    test "With valid parameters and without filters", %{product: product} do
+      assert {:ok, [product]} == ProductsService.list(%{})
+    end
+
+    test "with filters", %{product: product, list_valid: list_valid} do
+      with_mock(ElasticService, [],
+        filter_search: fn %{"name" => "Nome-teste"} -> {:ok, product} end
+      ) do
+        assert {:ok, product} == ProductsService.list(list_valid)
+      end
+    end
+
+    test "with invalid filters" do
+      with_mock(ElasticService, [],
+        filter_search: fn %{"amount" => "invalid"} ->
+          {:error, :bad_request, "For input string: \"invalid\""}
+        end
+      ) do
+        assert {:error, :bad_request, "For input string: \"invalid\""} ==
+                 ProductsService.list(%{"amount" => "invalid"})
+      end
+    end
+
+    test "Without index" do
+      ElasticService.delete_all()
```

Comment:
delete_all apaga o indice ou todos os produtos nele? Se for o indice inteiro esse teste não é necessário.